src/Order/OrderTests/APITests.cs

```diff
@@ -0,0 +1,246 @@
+			mediator.Setup(m => m.Send(It.IsAny<GetCartQuery>(), It.IsAny<CancellationToken>()))
+				.ReturnsAsync((CartDetails)null);
+			var linkGenerator = LinkGeneratorMock.GetLinkGeneratorMock();
+			var controller = GetController(mediator.Object, linkGenerator.Object);
+
+			var result = await controller.GetDetails(Guid.NewGuid());
+
+			Assert.IsType<NotFoundResult>(result);
+		}
+
+		[Fact]
+		public async Task GetDetails_CartItemsReturned_OkObjectResultWithCorrectResult()
+		{
+			var mediator = new Mock<IMediator>();
+			var cart = new CartDetails
+			{
+				Items =
+				[
+					new () { EventSeat = new()},
+					new () { EventSeat = new()},
+				]
+			};
+			mediator.Setup(m => m.Send(It.IsAny<GetCartQuery>(), It.IsAny<CancellationToken>()))
+				.ReturnsAsync(cart);
+			var linkGenerator = LinkGeneratorMock.GetLinkGeneratorMock();
+			var controller = GetController(mediator.Object, linkGenerator.Object);
+
+			var result = await controller.GetDetails(Guid.NewGuid());
+			var value = (result as OkObjectResult)?.Value as Resource<CartDetails>;
+
+			Assert.NotNull(value?.Value?.Items);
+			Assert.Equal(cart.Items.Count, value.Value.Items.Count);
+		}
+
+		[Fact]
+		public async Task AddToCart_ResultNull_NotFoundWithMessage()
+		{
+			var mediator = new Mock<IMediator>();
+			var guid = Guid.NewGuid();
+			mediator.Setup(m => m.Send(It.IsAny<AddItemToCartCommand>(), It.IsAny<CancellationToken>()))
+				.ReturnsAsync((Result<Unit>)null);
+			var linkGenerator = LinkGeneratorMock.GetLinkGeneratorMock();
+			var controller = GetController(mediator.Object, linkGenerator.Object);
+			var cartItem = new CartItemInputModel();
+
+			var result = await controller.AddToCart(cartItem, guid);
+			var value = (result as NotFoundObjectResult)?.Value as string;
+
+			Assert.NotNull(value);
+			Assert.NotEmpty(value);
+		}
+
+		[Fact]
+		public async Task AddToCart_ResultSuccess_OkWithCartDetails()
+		{
+			var mediator = new Mock<IMediator>();
+			var guid = Guid.NewGuid();
+			var successResult = new Result<Unit>
+			{
+				Error = false,
+				Value = Unit.Value
+			};
+			var cart = new CartDetails
+			{
+				Items =
+				[
+					new () { EventSeat = new()},
+				]
+			};
+			mediator.Setup(m => m.Send(It.IsAny<AddItemToCartCommand>(), It.IsAny<CancellationToken>()))
+				.ReturnsAsync(successResult);
+			mediator.Setup(m => m.Send(It.IsAny<GetCartQuery>(), It.IsAny<CancellationToken>()))
+				.ReturnsAsync(cart);
+			var linkGenerator = LinkGeneratorMock.GetLinkGeneratorMock();
+			var controller = GetController(mediator.Object, linkGenerator.Object);
+			var cartItem = new CartItemInputModel();
+
+			var result = await controller.AddToCart(cartItem, guid);
+			var value = (result as OkObjectResult)?.Value as Resource<CartDetails>;
+
+			Assert.NotNull(value?.Value?.Items);
+			Assert.Equal(cart.Items.Count, value.Value.Items.Count);
+		}
+
+		[Fact]
+		public async Task AddToCart_ResultSuccess_ResultNull_ThrowsNullReference()
+		{
```

Comment:
This looks weird - why controller throws? shouldn't API return IActionResult? what does it mean?

Reply:
I somehow forgot that we don't test unreal cases. Wouldn't it result in 500th error from api?